Bottom: 147cbb292a19cdaa49cb27217c9c1ae686469eef
Top:    e67eb4a93b7515014402ae0647d0a76dece141c3
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-03-18 20:39:13 +0000

Improve dump_poly() output

Make it print each contour of a POLYAREA, and move the NEXT PLINE
and NEXT POLY messages inside the loops interating over these
structures.


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 2ffd7cb..d9f6ce4 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -142,20 +142,26 @@ pline_dump (VNODE * v)
 	       n->point[0], n->point[1], theState (v));
     }
   while ((v = v->next) != s);
-  fprintf (stderr, "NEXT PLINE\n");
 }
 
 static void
 poly_dump (POLYAREA * p)
 {
   POLYAREA *f = p;
+  PLINE *pl;
 
   do
     {
-      pline_dump (&p->contours->head);
+      pl = p->contours;
+      do
+        {
+          pline_dump (&pl->head);
+          fprintf (stderr, "NEXT PLINE\n");
+        }
+      while ((pl = pl->next) != NULL);
+      fprintf (stderr, "NEXT POLY\n");
     }
   while ((p = p->f) != f);
-  fprintf (stderr, "NEXT_POLY\n");
 }
 #endif
